src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -316,10 +362,19 @@ def run_grace_tool(self,
             result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                                     timeout=10)
             if result.returncode:
+                stderr = result.stderr.decode("utf-8")
                 self.mgr.log.warning(
-                    f'ganesha-rados-grace tool failed: {result.stderr.decode("utf-8")}'
+                    f'ganesha-rados-grace tool failed for service {spec.service_name()}, '
```

Comment:
usage of f-strings in logging is not recommended, it's better to use normally formatting (using %) https://docs.astral.sh/ruff/rules/logging-f-string/